### DISCUSSION TITLE
CUDA our of memory error

### DISCUSSION THREAD

**Discussion by Ayanaminn** [openai/whisper#376] [Q&A] [answered]
Hi guys,

I'm using a Nvidia GeForce GTX 730. It is a pretty old GPU and only have 2GB memory. I was able to run "base" model without problem but start from the "small" model I got the error message:
 `torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 20.00 MiB (GPU 0; 2.00 GiB total capacity; 1.32 GiB already allocated; 0 bytes free; 1.40 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF`

Is there anything I can do to solve this problem? Thanks in advance.

**Comment by jianfch** [comment, marked as answer]
The small model is bigger than the base model so it requires more vram. You can run it on cpu instead of cuda.

**Reply by YSternlicht** [reply]
How would one run it directly on CPU instead of cuda?

**Reply by jianfch** [reply]
```python
import whisper
model = whisper.load_model('small', 'cpu')
```
or use argument `--device=cpu` for cli

**Reply by Ayanaminn** [reply]
Thank you. You are right 2GB is simply not enough to handle larger models. 
Use CPU can be an option but it just take too much time. I might just run it in colab.

**Comment by NorthSERG** [comment]
Guys! I have such problem. During the recursive check of empty folder if it has files or no I get the message "CUDA out of memory. Tried to allocate 26.00 MiB. GPU 0 has a total capacty of 2.00 GiB of which 0 bytes is free. Of the allocated memory 4.80 GiB is allocated by PyTorch, and 292.88 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF". But when I transcribe files in the folder I don't have this error. Device GTX960M, model Medium.
